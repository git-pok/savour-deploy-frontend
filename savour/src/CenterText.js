import './CenterText.css';

/**
 * CenterText
 * Props: text, color, fontSize
 * Renders a paragraph element.
*/
const CenterText = ({ text, color, fontSize }) => {
  const styles = {
    color,
    fontSize
  };
  return (
    <p style={styles}>{text}</p>
  );
}

export default CenterText;