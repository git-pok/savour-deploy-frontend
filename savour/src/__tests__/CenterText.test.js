import CenterText from '../CenterText.js';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

test("renders CenterText without crashing", () => {
  render(
    <MemoryRouter>
    <CenterText
      text="Test Text!"
      color="black"
      fontSize={24} />
    </MemoryRouter>);
});

test("renders snapshot without crashing", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <CenterText
        text="Test Text!"
        color="black"
        fontSize={24} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});
