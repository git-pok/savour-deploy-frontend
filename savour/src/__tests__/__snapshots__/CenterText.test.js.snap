// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot without crashing 1`] = `
<DocumentFragment>
  <p
    style="color: black; font-size: 24px;"
  >
    Test Text!
  </p>
</DocumentFragment>
`;
